### PR TITLE
Adding support for Python 2-3

### PR DESCRIPTION
**Pull request by holtgrewe**
This PR adds Python 2 and 3 support for OptiType using the `six` and `future` packages.

Also, Travis CI is configured to build and test for v3.*. The change starts out with the `futurize` results and then hand-fixing the remaining problems.

### CODE REVIEW

File: OptiTypePipeline.py [hunk 1]
```diff
@@ -245,7 +259,12 @@ def get_num_threads(configured_threads):
     out_dir = os.path.join(args.outdir, date)
     os.makedirs(out_dir)
 
-    bam_paths = args.input if bam_input else [os.path.join(out_dir, ("%s_%i.bam" % (date, i+1))) for i in range(len(args.input))]
+    if PYSAM_AVAILABLE:
+        extenion = 'bam'
```
Comment (b-schubert) [inline OptiTypePipeline.py]: misspelled extension. This causes Travis to fail.

**Comment by b-schubert** [comment]
Could you please rebase to master again to get the most recent changes as of today.

**Comment by holtgrewe** [comment]
I rebased this as asked for. The failure comes through some Docker-related addition independent after my changes.

**Comment by holtgrewe** [comment]
ping :bell:

**Comment by b-schubert** [comment]
You can rebase again. It should work now.

**Pull request merged**